flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobManagerRunnerResult.java

```diff
@@ -81,21 +83,23 @@ public boolean equals(Object o) {
             return false;
         }
         JobManagerRunnerResult that = (JobManagerRunnerResult) o;
-        return Objects.equals(archivedExecutionGraph, that.archivedExecutionGraph)
+        // TODO: verify ExecutionGraphInfo.equals
```

Comment:
still valid?

Reply:
I wanted to re-iterate over the `equals`/`hashCode` contract for `ExecutionGraphInfo` and `ArchivedExecutionGraph`. The `JobManagerRunnerResult` must have slipped through when going through the `TODO`s.

I was just wondering whether there should have been an `equals`/`hashCode` implementation for `ArchivedExecutionGraph` in the first place. Isn't `ArchivedExecutionGraph` well-defined by its `ArchivedExecutionJobVertex` instances stored in `ArchivedExecutionGraph::tasks`? And `ArchivedExecutionJobVertex` would be well-defined through its `ArchivedExecutionVertex` instances stored in `ArchivedExecutionJobVertex::taskVertices` which would be defined through its `ArchivedExecutionVertex::currentExecution`? I guess, we wouldn't gain anything code-wise besides having the contract formulated in the code which might increase readability. Any thoughts on this @zentol @tillrohrmann ?

Reply:
I think `JobManagerRunnerResult` should not have an `equals` and `hashCode` method. Usually only objects which you want to compare or use in a `HashMap` should implement these methods. We sometimes added these methods because we did not write a `Matcher` when using it in tests. However, this was more because of laziness than good software engineering. The problem is that if this is only used in tests, then it should probably be a test concern.

Reply:
I see, makes sense.